### PR TITLE
Deploy docs

### PR DESCRIPTION
**Pull request by imstar15**
Deploy rusts docs to https://oak.tech/rust when pushing code to master.

Refers to:
https://paritytech.github.io/substrate/master/

index.html will redirect to a library's document.

**Please set `AWS_ROLE_TO_ASSUME`, `SLACK_WEBHOOK_URL` in github action secrets.**

### CODE REVIEW

**Comment by imstar15** [comment]
Testing.

There is still an unresolved issue:
cargo doc doesn't generate index.html and I'm thinking about how to fix it.

File: scripts/build-docs/build-docs.sh [hunk 1]
```diff
@@ -0,0 +1,3 @@
+cargo doc
```
Comment (irsal) [inline scripts/build-docs/build-docs.sh:1]: can we only run this script if the lib.rs files in the relevant pallets are changed pls?

**Review by irsal — CHANGES_REQUESTED** [review]

File: scripts/build-docs/build-docs.sh (hunk 1 above)
Comment (irsal) [reply scripts/build-docs/build-docs.sh:1]: i don't want this to run during every merge

**Comment by irsal** [comment]
Process:
* detect if one of our pallet's lib.rs files changed
* merge master
* generate the rust docs post-merge and push to s3
* if any of this fails, post a slack notif

File: scripts/build-docs/build-docs.sh (hunk 1 above)
Comment (imstar15) [reply scripts/build-docs/build-docs.sh:1]: So code changes in runtime and node won't trigger uploading docs?

Comment (irsal) [reply scripts/build-docs/build-docs.sh:1]: code changes will do so, but only for pallet-*/src/lib.rs

**Comment by imstar15** [comment]
> Process:
> 
> * detect if one of our pallet's lib.rs files changed
> * merge master
> * generate the rust docs post-merge and push to s3
> * if any of this fails, post a slack notif

Yes, the process has been implemented. Please review.

@irsal

**Review by irsal — APPROVED** [review]
looks great, let's merge and test this

**Pull request merged**